src/middleware.ts

```diff
@@ -1,9 +1,12 @@
-import { authMiddleware } from "@clerk/nextjs";
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
+import { NextResponse } from "next/server";
 
-export default authMiddleware({
-  publicRoutes: ["/"],
-});
+const isProtectedDashboardRoute = createRouteMatcher([/dashboard.*/]);
+
+export default clerkMiddleware(async (auth, req) => {
+  if (isProtectedDashboardRoute(req) && !auth().userId) {
+    auth().protect();
```

Comment:
Does this need to be: 
```suggestion
    return auth().protect();
```

Reply:
Did we find out if its necessary to return the protect function?